portfolio/src/main/webapp/calendar.js

```diff
@@ -151,15 +152,180 @@ function collectAllEvents() {
     const startTime = new Date(eventCardBody.childNodes[1].innerText);
     const endTime = new Date(eventCardBody.childNodes[2].innerText);
     const event = new CalendarEvent(eventName, startTime, endTime);
-    return event; 
+    return event;
   });
-
+ 
   return allEventsOnPage;
 }
-
+ 
 /** Checks if two calendar events are identical. */
 function eventsEqual(eventA, eventB) {
   return (eventA.name === eventB.name) &&
       (eventA.startTime.getTime() === eventB.startTime.getTime()) &&
       (eventA.endTime.getTime() === eventB.endTime.getTime());
 }
+
+/** Retrives the date that the user has picked for the scheduling. */
+function getUserPickedDate() {
```

Comment:
Consider renaming to getUserPickedDateFromDom for clarity.

Reply:
Updated.